controller/space_template_blackbox_test.go

```diff
@@ -0,0 +1,378 @@
+
+	require.NotNil(t, *res.Data.Attributes.Description)
+	require.Equal(t, *spaceTemplate.Description, *res.Data.Attributes.Description)
+
+	require.NotNil(t, *res.Data.Attributes.Template)
+	require.Equal(t, *appSpaceTemplate.Attributes.Template, *res.Data.Attributes.Template)
+	return res
+}
+
+func (s *TestSpaceTemplateSuite) TestSpaceTemplate_Create() {
+	req := &goa.RequestData{
+		Request: &http.Request{Host: "api.service.domain.org"},
+	}
+
+	s.T().Run("invalid YAML template", func(t *testing.T) {
+		t.Parallel()
+		// given
+		svc, ctrl := s.SecuredController()
+		spaceTemplate := spacetemplate.SpaceTemplate{
+			Name:     uuid.NewV4().String(), // in order to create a random space template name
+			Template: "this is not a valid template ;)",
+		}
+		var jerr *app.JSONAPIErrors
+		// when
+		application.Transactional(s.db, func(appl application.Application) error {
+			payload := &app.CreateSpaceTemplatePayload{
+				Data: ConvertSpaceTemplate(appl, req, spaceTemplate),
+			}
+			_, jerr = test.CreateSpaceTemplateBadRequest(t, svc.Context, svc, ctrl, payload)
+			return nil
+		})
+		// then
+		require.NotNil(t, jerr)
+	})
+
+	s.T().Run("empty template (valid YAML)", func(t *testing.T) {
+		t.Parallel()
+		// given
+		svc, ctrl := s.SecuredController()
+		spaceTemplate := spacetemplate.SpaceTemplate{
+			Name:     uuid.NewV4().String(), // in order to create a random space template name
+			Template: "test: demo",          // this is valid YAML but not a valid template which is fine
+		}
+		var createdSpaceTemplate *app.SpaceTemplateSingle
+		// when
+		application.Transactional(s.db, func(appl application.Application) error {
+			payload := &app.CreateSpaceTemplatePayload{
+				Data: ConvertSpaceTemplate(appl, req, spaceTemplate),
+			}
+			_, createdSpaceTemplate = test.CreateSpaceTemplateCreated(t, svc.Context, svc, ctrl, payload)
+			return nil
+		})
+		// then
+		require.NotNil(t, createdSpaceTemplate)
+	})
+
+	s.T().Run("bare minimum template", func(t *testing.T) {
+		t.Parallel()
+		_ = s.createMinimumSpaceTemplate(t)
+	})
+
+	s.T().Run("unauthorized", func(t *testing.T) {
+		t.Parallel()
+		// given
+		svc, ctrl := s.UnSecuredController()
+		spaceTemplate := spacetemplate.SpaceTemplate{
+			Name:     uuid.NewV4().String(), // in order to create a random space template name
+			Template: "test: demo",          // this is valid YAML but not a valid template which is fine
+		}
+		var jerrs *app.JSONAPIErrors
+		// when
+		application.Transactional(s.db, func(appl application.Application) error {
+			payload := &app.CreateSpaceTemplatePayload{
+				Data: ConvertSpaceTemplate(appl, req, spaceTemplate),
+			}
+			_, jerrs = test.CreateSpaceTemplateUnauthorized(t, svc.Context, svc, ctrl, payload)
+			return nil
+		})
+		// then
+		require.NotNil(t, jerrs)
+	})
+}
+
+func (s *TestSpaceTemplateSuite) TestSpaceTemplate_Show() {
+	s.T().Run("show non-existing template", func(t *testing.T) {
+		t.Parallel()
```

Comment:
If you don't call it in the subtests, they won't be executed in parallel but sequentially. See *Run a group of tests in parallel* section [here](https://blog.golang.org/subtests).